lib/webResource.ts

```diff
@@ -167,11 +168,22 @@ export class WebResource {
             }
           }
         }
-      }// end-of-for
+      } // end-of-for
       // append the queryString
       this.url += queryParams.join("&");
     }
 
+    // set formData parameters for 'application/x-www-form-urlencoded' or 'multipart/form-data'.
```

Comment:
This is true only for the request.js library. https://github.com/request/request#requestoptions-callback. and search for `application/x-www-form-urlencoded`. You would see that the request library takes form and formData as separate options. Hence I made this change in the node sdk's ms-rest. Here we are using browser's fetch or node-fetch and we need to find out what the underlying library does when content-type is `"application/x-www-form-urlencoded"` v/s `"multipart/form-data"`.

Reply:
So basically setting a `form` property on the request object was just because you know the request.js lib will go looking there for it?

Reply:
yup. And I also added a request filter over here https://github.com/Azure/azure-sdk-for-node/pull/2366/files

Reply:
OK, will sort this out before merge

Reply:
This is the issue that I created https://github.com/Azure/ms-rest-js/issues/8. I also provided links to similat issues that other people reported in those repos. This will be helpful. To provide context: I had to make this change in ms-rest for node sdk because cognitive services needed this.

Reply:
I think this is a very big issue on it's own right. For now you can probably revert the changes related to formData and solve them as a part of solving #8.

Reply:
Pushing the formData commit to a different branch to handle later.